cryptographic_estimators/UOVEstimator/UOVAlgorithms/direct_attack.py

```diff
@@ -20,31 +20,27 @@
 from ..uov_problem import UOVProblem
 from ...MQEstimator.mq_estimator import MQEstimator
 from ...MQEstimator.MQAlgorithms.lokshtanov import Lokshtanov
-from ...base_constants import BASE_MEMORY_BOUND, BASE_BIT_COMPLEXITIES, BASE_EXCLUDED_ALGORITHMS
-from math import log2
+from ...base_constants import BASE_EXCLUDED_ALGORITHMS
 from cryptographic_estimators.base_constants import BASE_FORGERY_ATTACK
 
 class DirectAttack(UOVAlgorithm):
-    """
-    Construct an instance of DirectAttack estimator
-
-    The most straightforward attack against UOV, (and even against most of the MPKC 
-    cryptosystems) is the direct attack, where the attacker aims to solve an instance of the 
-    MQ problem associated with the public key [TW12]_.
-
-    INPUT:
-
-    - ``problem`` -- an instance of the UOVProblem class
-    - ``w`` -- linear algebra constant (default: 2)
-    - ``h`` -- external hybridization parameter (default: 0)
-    - ``excluded_algorithms`` -- a list/tuple of MQ algorithms to be excluded (default: [Lokshtanov])
-    - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
-    - ``complexity_type`` -- complexity type to consider (0: estimate, 1: tilde O complexity, default: 0)
-    - ``bit_complexities`` -- determines if complexity is given in bit operations or basic operations (default 1: in bit)
-
-    """
-
     def __init__(self, problem: UOVProblem, **kwargs):
+        """Construct an instance of DirectAttack estimator.
+
+        The most straightforward attack against UOV, (and even against most of the MPKC 
+        cryptosystems) is the direct attack, where the attacker aims to solve an instance of the 
+        MQ problem associated with the public key [TW12]_.
+
+        Args:
+            problem (UOVProblem): An instance of the UOVProblem class.
+            **kwargs: Additional keyword arguments.
+                w (int): Linear algebra constant (default: 2).
+                h (int): External hybridization parameter (default: 0).
+                excluded_algorithms (list): A list/tuple of MQ algorithms to be excluded (default: [Lokshtanov]).
+                memory_access (int): Specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage).
+                complexity_type (int): Complexity type to consider (0: estimate, 1: tilde O complexity, default: 0).
+                bit_complexities (int): Determines if complexity is given in bit operations or basic operations (default 1: in bit).
```

Comment:
wrong indentation

Reply:
to be fixed in  #198

Reply:
Nothing to fix. Indentation is correctly parsed by our custom Sphinx extension (see https://github.com/Crypto-TII/CryptographicEstimators/pull/211).